liblumen_codegen/lib/lumen/compiler/Translation/ModuleBuilder.cpp

```diff
@@ -715,23 +741,104 @@ Value ModuleBuilder::build_logical_or(Value lhs, Value rhs) {
   return op.getResult();
 }
 
-extern "C" MLIRValueRef MLIRBuildPrintOp(MLIRModuleBuilderRef b,
-                                         MLIRValueRef *argv, unsigned argc) {
-  ModuleBuilder *builder = unwrap(b);
-  SmallVector<Value, 1> args;
-  unwrapValues(argv, argc, args);
-  return wrap(builder->build_print_op(args));
+//===----------------------------------------------------------------------===//
+// Function Calls
+//===----------------------------------------------------------------------===//
+
+#define INTRINSIC_BUILDER(Alias, Op)                             \
+  static Value Alias(OpBuilder &builder, ArrayRef<Value> args) { \
+    auto op = builder.create<Op>(builder.getUnknownLoc(), args); \
+                                                                 \
+    return op.getResult();                                       \
+  }
+
+INTRINSIC_BUILDER(buildIntrinsicPrintOp, PrintOp);
+INTRINSIC_BUILDER(buildIntrinsicAddOp, AddOp);
+INTRINSIC_BUILDER(buildIntrinsicSubOp, SubOp);
+INTRINSIC_BUILDER(buildIntrinsicMulOp, MulOp);
+INTRINSIC_BUILDER(buildIntrinsicDivOp, DivOp);
+INTRINSIC_BUILDER(buildIntrinsicRemOp, RemOp);
+INTRINSIC_BUILDER(buildIntrinsicFDivOp, FDivOp);
+INTRINSIC_BUILDER(buildIntrinsicBslOp, BslOp);
+INTRINSIC_BUILDER(buildIntrinsicBsrOp, BsrOp);
+INTRINSIC_BUILDER(buildIntrinsicBandOp, BandOp);
+// INTRINSIC_BUILDER(buildIntrinsicBnotOp, BnotOp);
```

Comment:
Was there a problem with `not` or just wasn't needed yet?

Reply:
It's a unary operator and I only implemented the generic class for the binary operators - easy enough to add support for it, just didn't do that yet.